app/src/main/java/com/example/myapplication/adapter/UsersAdapter.kt

```diff
@@ -19,62 +18,58 @@ private const val MULTISELECT_MODE_MARGIN = 40
-    private val isModeActive: LiveData<Boolean> get() = _isModeActive
+    private var isModeActive: Boolean = false
 
+    @SuppressLint("NotifyDataSetChanged")
     fun changeModeStatus(state: Boolean) {
-        _isModeActive.value = state
+        isModeActive = state
+        notifyDataSetChanged()
     }
 
-    class UsersViewHolder(
+    inner class UsersViewHolder(
         private val binding: UserPatternBinding,
-        private val actionListener: UserActionListener,
-        private val isModeActive: LiveData<Boolean>,
-        private val setModeActive: (Boolean) -> Unit
     ) : RecyclerView.ViewHolder(binding.root) {
 
         fun onBind(item: User) {
             val context = binding.root.context
             val density = context.resources.displayMetrics.density
 
-            isModeActive.observeForever { active ->
-                with(binding) {
-                    val layoutParams =
-                        userPhotoView.layoutParams as ViewGroup.MarginLayoutParams
-                    root.setBackgroundResource(
-                        if (active) R.drawable.selected_user_background
-                        else R.drawable.item_background_selector
-                    )
-                    layoutParams.marginStart =
-                        if (active) (MULTISELECT_MODE_MARGIN * density).toInt()
-                        else (DEFAULT_MARGIN * density).toInt()
-                    userPhotoView.layoutParams = layoutParams
-                    deleteUserView.visibility = if (active) View.GONE else View.VISIBLE
-                    checkBox.visibility = if (active) View.VISIBLE else View.GONE
-                    checkBox.isChecked = item.isSelected
+            with(binding) {
+                val layoutParams =
+                    userPhotoView.layoutParams as ViewGroup.MarginLayoutParams
+                root.setBackgroundResource(
+                    if (isModeActive) R.drawable.selected_user_background
+                    else R.drawable.item_background_selector
+                )
+                layoutParams.marginStart =
+                    if (isModeActive) (MULTISELECT_MODE_MARGIN * density).toInt()
+                    else (DEFAULT_MARGIN * density).toInt()
+                userPhotoView.layoutParams = layoutParams
+                deleteUserView.visibility = if (isModeActive) View.GONE else View.VISIBLE
+                checkBox.visibility = if (isModeActive) View.VISIBLE else View.GONE
+                checkBox.isChecked = item.isSelected
 
-                    userNameView.text = item.name
-                    userCareerView.text = item.career
+                userNameView.text = item.name
+                userCareerView.text = item.career
 
-                    itemView.setOnLongClickListener {
-                        setModeActive(true)
-                        actionListener.onMultiSelectModeActive()
+                itemView.setOnLongClickListener {
```

Comment:
You have clicklistener for itemView, but you do not have clicklistener for checkbox, and so list cannot return to its default state if you clear all the checkboxes by intentionally clicking on the checkboxes rather than the itemView.